jcl/src/jdk.attach/share/classes/com/ibm/tools/attach/attacher/OpenJ9AttachProvider.java

```diff
@@ -81,13 +86,37 @@ public OpenJ9VirtualMachine attachVirtualMachine (
 		}
 
 		OpenJ9VirtualMachine vm = new OpenJ9VirtualMachine(this, descriptor.id());
-		vm.attachTarget();
+		PrivilegedExceptionAction<Object> action = () -> {
+			vm.attachTarget();
```

Comment:
Since attachTarget() is modified to use doPriv, I don't expect this one is needed.

Reply:
That is true, will remove this doPriv.

Reply:
Removed this `doPrivileged()`.
In addition, modified the testcase https://github.com/eclipse/openj9/issues/6119#issuecomment-504538933 to invoke following public APIs:
```
	vm.getAgentProperties();
	vm.provider();
	vm.startLocalManagementAgent();
	vm.startManagementAgent(props);
	vm.list();
``` 
There was no other `AccessControlException` discovered.

Ideally, existing `AttachAPISanity` test should have a test mode with security manager enabled, will leave that to a separated PR though.

Please review.

Reply:
Probably they are fine as well, but there is also the following to check:
detach()
loadAgent()
loadAgentLibrary()
loadAgentPath()
executeDiagnosticCommand()
heapHisto()

Reply:
Good call. Tested following public APIs in modified testcase https://github.com/eclipse/openj9/issues/6119#issuecomment-504538933:
```
	j9vm.getThreadInfo();
	j9vm.heapHisto("-all");
	j9vm.executeDiagnosticCommand("live");
```
Following APIs are tested via `make _TestAttachAPI (org.openj9.test.attachAPI.TestAttachAPI)`:
```
vm.loadAgentLibrary(JVMTITST);
vm.loadAgentLibrary(JVMTITST, "test=aot001");
vm.loadAgentPath(agent);
vm.loadAgentPath(agent, options);
```
`vm.detach()` has already been tested.

Added `doPrivileged()` for `OpenJ9VirtualMachine.heapHisto()`. Please have another look.

Reply:
Not sure it matters, but "live" isn't  a valid diagnostic command. Valid command appear to be:
"Thread.print"
"GC.run"
"GC.class_histogram"

Reply:
`live` is a modifier for `heapHisto`.  It causes a `GC.run` before `GC.class_histogram`.

Reply:
Those `diagnostic` commands didn't make difference. However just discovered another system property retrieval for `openj9.tools.attach.diagnostics.debug`, which was marked by an excessive custom policy file.